config/src/config/config_optimizer.rs

```diff
@@ -4,17 +4,47 @@
 use crate::{
     config::{
         node_config_loader::NodeType, utils::get_config_name, Error, InspectionServiceConfig,
-        LoggerConfig, MempoolConfig, NodeConfig, PeerMonitoringServiceConfig, StateSyncConfig,
+        LoggerConfig, MempoolConfig, NodeConfig, Peer, PeerMonitoringServiceConfig, PeerRole,
+        PeerSet, StateSyncConfig,
     },
     network_id::NetworkId,
 };
-use aptos_types::chain_id::ChainId;
+use aptos_crypto::{x25519, ValidCryptoMaterialStringExt};
+use aptos_types::{chain_id::ChainId, network_address::NetworkAddress, PeerId};
+use maplit::hashset;
 use serde_yaml::Value;
+use std::{collections::HashMap, str::FromStr};
 
 // Useful optimizer constants
 const OPTIMIZER_STRING: &str = "Optimizer";
+const PUBLIC_NETWORK_OPTIMIZER_NAME: &str = "PublicNetworkConfigOptimizer";
 const VALIDATOR_NETWORK_OPTIMIZER_NAME: &str = "ValidatorNetworkConfigOptimizer";
 
+// Mainnet seed peer addresses and keys
+const MAINNET_SEED_PEER_ADDRESSES: [&str; 0] = [];
+const MAINNET_SEED_PEER_NETWORK_ADDRESSES: [&str; 0] = [];
+const MAINNET_SEED_PEER_PUBLIC_KEYS: [&str; 0] = [];
```

Comment:
It would be better to use a more specific type that 

 1) Ties together these 3 properties
 2) Parses the strings into something more appropriate, like a bignum

Reply:
The address is not a string, it's a large integer, so it should be parsed as such. The string is only its printed representation.

Reply:
Aah, gotcha 😄
1. The peer address here is a hex formatted string. When we build the seed from it, we use `PeerId::from_hex(...)`, (which appears to be the standardized way of handling addresses across our code). Likewise, it's the format expected by the serde implementation for the object (e.g., if you were to define these in the node config manually).
2. I also think it's much more readable than storing a large number in the code. For example, I can directly look it up in our [explorer](https://explorer.aptoslabs.com/account/31e55012a7d439dcd16fee0509cd5855c1fbdc62057ba7fac3f7c88f5453dd8e?network=testnet) 😄  

So, I'm inclined to leave this as it is. I think it's superior to a large number 🤔